app/(map)/_components/ProfileCardItem.tsx

```diff
@@ -13,24 +16,40 @@ import {
 } from "@/app/_common/shadcn/ui/card";
 import { Badge as Tag } from "@/app/_common/shadcn/ui/badge";
 
-import { Creator } from "@/app/_common/types/profile";
+import { Profile } from "@/app/_common/types/profile";
+
+import useToggleLikeProfile from "../_api/useToggleLikeProfile";
 
 interface Props {
-  profile: Creator;
+  profile: Profile;
 }
 
 function ProfileCardItem({ profile }: Props) {
   const {
-    username,
-    githubId,
-    avatarUrl,
-    githubUrl,
-    bio,
-    jandiRate,
-    achievementTitle,
-    developLanguages,
-    wantedJobs,
+    response: {
+      id,
+      username,
+      githubId,
+      avatarUrl,
+      githubUrl,
+      bio,
+      jandiRate,
+      achievementTitle,
+      developLanguages,
+      wantedJobs,
+    },
+    requestYn,
   } = profile;
+  const { user } = useAuthStore();
+  const { toggleLikeProfile, isLiked } = useToggleLikeProfile();
+
+  const handleToggleButton = () => {
+    if (!user) return;
+    toggleLikeProfile({
+      isLiked: isLiked ?? requestYn,
+      likeInfo: { senderId: user.id, receiverId: id },
```

Comment:
```suggestion
      id: receiverId,
      username,
      githubId,
      avatarUrl,
      githubUrl,
      bio,
      jandiRate,
      achievementTitle,
      developLanguages,
      wantedJobs,
    },
    requestYn,
  } = profile;
  const { user } = useAuthStore();
  const { toggleLikeProfile, isLiked } = useToggleLikeProfile();

  const handleToggleButton = () => {
    if (!user) return;
    toggleLikeProfile({
      isLiked: isLiked ?? requestYn,
      likeInfo: { senderId: user.id, receiverId },
```

이렇게 변경하는게 더 직관적으로 무슨 id인지 알아볼 것 같아서 제안드려요~